In [2]:
import numpy as np

In [5]:
def cos(deg: float) -> float:
    return np.cos(np.radians(deg))

def sin(deg: float) -> float:
    return np.sin(np.radians(deg))

## Resistive/Reactive Load

![resistive-reactive-load](https://www.allaboutcircuits.com/uploads/articles/true-reactive-apparent-power-for-a-purely-resistive-reactive-load.png)

An identical circuit was described by the following equations in the [first section](ResistiveAndReactive.ipynb#AC-circuit-with-a-Resistive-and-Purely-Reactive-load) of this chapter:

The reactance of the inductor and the total impedance of the circuit are given by:

$$ X_\mathrm{L} = 120\pi\mathrm{Hz} \cdot 0.160\mathrm{H} = 19.2\pi\,\Omega$$

\begin{align*}
    Z &= (60+19.2\pi j)\,\Omega \\
     &\approxeq (60 + 60.32j)\,\Omega \\
     &\approxeq 85.08\,\Omega\,\angle\,45.15^\circ
\end{align*}     

In [3]:
19.2*np.pi

60.31857894892403

And, the current is given by:  

$$ I = \frac{120\mathrm{V}}{Z} \approxeq 1.41\mathrm{A}\,\angle\!-45.15^\circ$$

As for (apparent) power, denoted as '$S$':  

$$ S = IE \approxeq 120\mathrm{V}\,\sin(\omega t) \cdot 1.41\mathrm{A}\,\sin\big(\omega t - 45.15^\circ \big)$$

Via the above equation, equation (3) from the [Trigonometry](/notebooks/Math/Trigonometry.ipynb) notebook, and substitution,  

\begin{align*}
    S &\approxeq 120\mathrm{V}\cdot1.41\mathrm{A}\cdot\sin(\omega t) \,\big(\sin(\omega t)\,\cos45.15^\circ \:-\: \cos(\omega t)\,\sin45.15^\circ \big) \\
     &\approxeq 169.2\mathrm{W}\,\big(0.705\,\sin^2(\omega t) - 0.709\,\sin(\omega t)\cos(\omega t)\big) \\
     &\approxeq 169.2\mathrm{W}\,\big(0.705\,\sin^2(\omega t) - 0.354\,\sin(2\omega t)\big) \\
     &\approxeq 169.2\mathrm{W}\,\Big(0.353\,\big(1 - \cos(2\omega t)\big) \:-\:0.354\,\sin(2\omega t)\Big) 
\end{align*}     

In [7]:
120**2*cos(45.15)/np.sqrt(60**2 + (19.2*np.pi)**2)

119.368108927009

In [8]:
120**2*sin(45.15)/np.sqrt(60**2 + (19.2*np.pi)**2)

119.99476088646807

More generally speaking,  

$$ S = |E||I|\,\sin(\omega t)\,\big(\sin(\omega t)\,\cos\theta \:+\: \cos(\omega t)\,\sin\theta\big)$$

Here, '$\theta$' denotes the phase angle of the current ($I$).

Via the above equation in conjunction with the identities of both the true power ($P$) and the reactive power ($Q$),

\begin{align*}
    S &= P\,\sin(\omega t) + Q\,\cos(\omega t) \\ 
     &= |P|\,\sin^2(\omega t) + |Q|\,\sin(\omega t)\cos(\omega t) \\
     &= 0.5\,\Big(|P|\,\big(1 - \cos(2\omega t)\big) + |Q|\,\sin(2\omega t) \Big) \\
     &= 0.5\,\Big(|P| - |P|\sin\big(2\omega t + 90^\circ\big) + |Q|\,\sin(2\omega t) \Big)
\end{align*}     

In terms of scalar values,  

\begin{align*}
    |S| &= |E||I| \approxeq 169.2\mathrm{VA} \\
    |P| &= |S|\cos\theta \approxeq 119.37 \mathrm{W}\\
    |Q| &= |S|\sin\theta \approxeq 119.99 \mathrm{VAR}
\end{align*}    

As for vector values,

$$ P:=P(t) = |P|\sin(\omega t)$$

$$ Q:=Q(t) = |Q|\sin(\omega t)$$

Now, more specifically,

$$ S \approxeq 59.73\mathrm{W}\,\big(1 - \sin\big(2\omega t + 90^\circ \big) - \sin(2\omega t)\big)$$

The Fourier Transform of $S$ has a dc component or level of $59.73\mathrm{W}$, wihch gives the center line of the wave, assuming a symmetric waveform, and one harmonic that can be expressed as a complex number.  

$$H_1\approxeq -59.73\mathrm{W}\,(1 + j)= -59.73\mathrm{W}\sqrt{2}\,\angle\,45^\circ\approxeq -84.47\mathrm{W}\,\angle\,45^\circ$$

The apparent power as a function of time can be expressed in real terms as:  

\begin{align*}
    S:=S(t) &\approxeq 59.73\mathrm{W} - 59.73\mathrm{W}\sqrt{2}\,\sin\big(2\omega t + 45^\circ\big) \\
     &= 59.73\mathrm{W}\,\Big(1 - \sqrt{2}\,\sin\big(2\omega t + 45^\circ\big) \Big)
\end{align*}

**In general**, assuming a sine wave, the DC level is equal to $0.5|P|$.  Furthermore, the smaller the DC level is, the more power is given back to the system, which means less energy is needed to rotate the shaft of the generator, assuming the power supply actually is a generator.  What's more, the relationship mentioned in the previous section of this chapter between $\theta$ and the ratio between the power absorbed by the load and the power released back into the system by the load is that the closer the complex number representation of either the impedance or the current is to the imaginary axis, the smaller the DC level is, which implies less power absorbtion.  This relationship is illustrated in the next section of this notebook.

Again, assuming a sine wave, the first and only harmonic of the Fourier Transform of $S$ is, in complex number form, 

\begin{align*}
    H_1 &= -0.5\,\big(|Q|\angle\,0^\circ + |P|\angle\,90^\circ \big) \\
     &= -0.5(|Q| + j|P|) \\
     &= -0.5|S|\,(\sin\theta + j\,\cos\theta)
\end{align*}     

Thus,

$$ H_1 = -0.5|S|\angle\big(90^\circ - \theta \big)$$